pkg/api/dashboard.go

```diff
@@ -288,13 +288,24 @@ func (hs *HTTPServer) PostDashboard(c *models.ReqContext, cmd models.SaveDashboa
 	var err error
 	cmd.OrgId = c.OrgId
 	cmd.UserId = c.UserId
+
 	trimDefaults := c.QueryBoolWithDefault("trimdefaults", false)
 	if trimDefaults && !hs.LoadSchemaService.IsDisabled() {
 		cmd.Dashboard, err = hs.LoadSchemaService.DashboardApplyDefaults(cmd.Dashboard)
 		if err != nil {
 			return response.Error(500, "Error while applying default value to the dashboard json", err)
 		}
 	}
+	// if folderID not given in request search for it with folderUID
+	if cmd.FolderId == 0 && cmd.FolderUID != "" {
+		s := dashboards.NewFolderService(c.OrgId, c.SignedInUser, hs.SQLStore)
+		folder, err := s.GetFolderByUID(cmd.FolderUID)
+		if err != nil {
+			return response.Error(500, "Error while getting the folder by folderUID", err)
```

Comment:
response.Error(http.StatusInternalServerError, "Error while getting the folder by folderUID") use the http package const is better.